docs/ha-considerations.md

```diff
@@ -0,0 +1,337 @@
+- `${PRIORITY}` should be higher on the master than on the backups. Hence `101` and `100` respectively will suffice.
+- `${AUTH_PASS}` should be the same for all `keepalived` cluster hosts, e.g. `42`
+- `${APISERVER_VIP}` is the virtual IP address negotiated between the `keepalived` cluster hosts.
+
+### haproxy configuration
+
+The `haproxy` configuration consists of one file: the service configuration file which is assumed to reside in a _/etc/haproxy_ directory. Note that however some Linux distributions may keep them elsewhere. The following configuration has been successfully used with `haproxy` version 2.1.3:
+
+```bash
+# /etc/haproxy/haproxy.cfg
+#---------------------------------------------------------------------
+# Global settings
+#---------------------------------------------------------------------
+global
+    log /dev/log local0
+    log /dev/log local1 notice
+    daemon
+
+#---------------------------------------------------------------------
+# common defaults that all the 'listen' and 'backend' sections will
+# use if not designated in their block
+#---------------------------------------------------------------------
+defaults
+    mode                    http
+    log                     global
+    option                  httplog
+    option                  dontlognull
+    option http-server-close
+    option forwardfor       except 127.0.0.0/8
+    option                  redispatch
+    retries                 1
+    timeout http-request    10s
+    timeout queue           20s
+    timeout connect         5s
+    timeout client          20s
+    timeout server          20s
+    timeout http-keep-alive 10s
+    timeout check           10s
+
+#---------------------------------------------------------------------
+# apiserver frontend which proxys to the masters
+#---------------------------------------------------------------------
+frontend apiserver
+    bind *:${APISERVER_DEST_PORT}
+    mode tcp
+    option tcplog
+    default_backend apiserver
+
+#---------------------------------------------------------------------
+# round robin balancing for apiserver
+#---------------------------------------------------------------------
+backend apiserver
+    option httpchk GET /healthz
+    http-check expect status 200
+    mode tcp
+    option ssl-hello-chk
+    balance     roundrobin
+        server ${HOST1_ID} ${HOST1_ADDRESS}:${APISERVER_SRC_PORT} check
+        # [...]
+```
+Again, there are some placeholders in `bash` variable style to expand:
+- `${APISERVER_DEST_PORT}` the port through which Kubernetes will talk to the API Server.
+- `${APISERVER_SRC_PORT}` the port used by the API Server instances
+- `${HOST1_ID}` a symbolic name for the first load-balanced API Server host
+- `${HOST1_ADDRESS}` a resolvable address (DNS name, IP address) for the first load-balanced API Server host
+- additional `server` lines, one for each load-balanced API Server host
+
+### Option 1: Run the services on the operating system
```

Comment:
should this and Option 2 be `####`?

Reply:
> should this and Option 2 be `####`?

I have decided to put the two examples on level ## to avoid too deep nesting. Hence here level ### would be corrrect. 
I think this makes sense because it avoids too deep nesting.